tools/nrtstress/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"flag"
+	"math/rand"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/nrtupdater"
+	"github.com/k8stopologyawareschedwg/resource-topology-exporter/pkg/nrtupdater/fake"
+)
+
+func main() {
+	var hostname string
+	var tmPolicy string
+	var interval time.Duration
+	var seed int
+	var dryRun bool
+	flag.StringVar(&hostname, "hosthame", "fake-host-0", "fake host name (not validated)")
+	flag.StringVar(&tmPolicy, "tm-policy", "best-effort", "topology manager policy (not validated)")
+	flag.DurationVar(&interval, "interval", 10*time.Second, "periodic update interval")
+	flag.IntVar(&seed, "random-seed", 0, "random seed (use time)")
+	flag.BoolVar(&dryRun, "dry-run", false, "don't send data")
+
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	nrtupdaterArgs := nrtupdater.Args{
+		Hostname:  hostname,
+		NoPublish: dryRun,
+	}
+
+	var randSeed int64 = time.Now().UnixNano()
+	if seed != 0 {
+		randSeed = int64(seed)
+	}
+
+	klog.Infof("seed: %v", randSeed)
+
+	rand.Seed(randSeed)
+
+	klog.Infof("generating fake periodic updates every %v", interval)
+
+	gen := fake.NewGenerator(interval)
+	go gen.Run()
+
+	klog.Infof("using NRT Updater args: %+#v", nrtupdaterArgs)
+
+	upd := nrtupdater.NewNRTUpdater(nrtupdaterArgs, tmPolicy)
+	upd.Run(gen.Infos, nil)
```

Comment:
I would expect the generator to send the data to some 'real' updater which is part of the existing stack.
I don't understand how sending fake data to a fake updater can be helpful. 
What am I missing here?

Reply:
I should have given more context in the commit message/PR cover letter, I'll fix that.
This tool want to stress the apiserver part; we can easily generate fake "realistic" NRT data, so we can easily assess the apiserver load. Generating fake data from the kubelet side requires more effort (https://github.com/k8stopologyawareschedwg/resource-topology-exporter/issues/84)